test/Feature/HLSLLib/refract.16.test

```diff
@@ -0,0 +1,193 @@
+    Format: Float16
+    Stride: 8
+    Data: [ 0x3800, 0x3800, 0x39a8, 0x0 ]
+    # 0.5, 0.5, 0.70703125, 0.0
+  - Name: Wall4D
+    Format: Float16
+    Stride: 8
+    Data: [ 0x3800, 0x3800, 0x3800, 0x3800 ]
+    # 0.5, 0.5, 0.5, 0.5
+  - Name: Result2D
+    Format: Float16
+    Stride: 4
+    ZeroInitSize: 16
+  - Name: ExpectedResult2D
+    Format: Float16
+    Stride: 4
+    Data: [ 0xb4d9, 0xbb6d, 0xb4d9, 0xbb6d, 0x0, 0x0, 0x0, 0x0 ]
+    # -0.30297852, -0.92822266, -0.30297852, -0.92822266, 0.0, 0.0, 0.0, 0.0,
+  - Name: Result3D
+    Format: Float16
+    Stride: 8
+    ZeroInitSize: 16
+  - Name: ExpectedResult3D
+    Format: Float16
+    Stride: 8
+    Data: [ 0xb607, 0xba04, 0xb817, 0x0, 0xb607, 0xba04, 0xb817, 0x0 ]
+    # -0.37670898, -0.7519531, 0.51123047, 0.0, -0.37670898, -0.7519531, 0.51123047, 0.0
+  - Name: Result4D
+    Format: Float16
+    Stride: 8
+    ZeroInitSize: 16
+  - Name: ExpectedResult4D
+    Format: Float16
+    Stride: 8
+    Data: [ 0xb400, 0xb900, 0xb000, 0xba00, 0xb400, 0xb900, 0xb000, 0xba00 ]
+    # -0.25, -0.625, -0.125, -0.75, -0.25, -0.625, -0.125, -0.75
+Results:  
+  - Result: CheckResult2D
+    Rule: BufferFloatULP
+    ULPT: 2
+    Actual: Result2D
+    Expected: ExpectedResult2D
+  - Result: CheckResult3D
+    Rule: BufferFloatULP
+    ULPT: 2
+    Actual: Result3D
+    Expected: ExpectedResult3D
+  - Result: CheckResult4D
+    Rule: BufferFloatULP
+    ULPT: 2
+    Actual: Result4D
+    Expected: ExpectedResult4D
+
+DescriptorSets:
+  - Resources:
+    - Name: IncidentRay2D
+      Kind: StructuredBuffer
+      DirectXBinding:
+        Register: 0
+        Space: 0
+      VulkanBinding:
+        Binding: 0
+    - Name: IncidentRay3D
+      Kind: StructuredBuffer
+      DirectXBinding:
+        Register: 1
+        Space: 0
+      VulkanBinding:
+        Binding: 1
+    - Name: IncidentRay4D
+      Kind: StructuredBuffer
+      DirectXBinding:
+        Register: 2
+        Space: 0
+      VulkanBinding:
+        Binding: 2
+    - Name: Wall2D
+      Kind: StructuredBuffer
+      DirectXBinding:
+        Register: 3
+        Space: 0
+      VulkanBinding:
+        Binding: 3
+    - Name: Wall3D
+      Kind: StructuredBuffer
+      DirectXBinding:
+        Register: 4
+        Space: 0
+      VulkanBinding:
+        Binding: 4
+    - Name: Wall4D
+      Kind: StructuredBuffer
+      DirectXBinding:
+        Register: 5
+        Space: 0
+      VulkanBinding:
+        Binding: 5
+    - Name: Result2D
+      Kind: RWStructuredBuffer
+      DirectXBinding:
+        Register: 6
+        Space: 0
+      VulkanBinding:
+        Binding: 6
+    - Name: Result3D
+      Kind: RWStructuredBuffer
+      DirectXBinding:
+        Register: 7
+        Space: 0
+      VulkanBinding:
+        Binding: 7
+    - Name: Result4D
+      Kind: RWStructuredBuffer
+      DirectXBinding:
+        Register: 8
+        Space: 0
+      VulkanBinding:
+        Binding: 8      
+...
+#--- end
+# UNSUPPORTED: Clang-Vulkan

```

Comment:
I think we settled on wanting to use `XFAIL` instead of `UNSUPPORTED` so that we do track when support comes in.